ONSdigital/dp-dataset-api
# upgraded dp-mongodb to version 1.8.0 to use latest locking algorithm

### What

- Upgrade dependencies to latest versions
- dp-mongodb: use new locking algorithm
- provide caller identity as unique caller name to the locking algorithm

### How to review

- Make sure code changes make sense
- Make sure unit tests and component test pass
- (info) tested locally with postman:
  - add a sleep of 3 seconds just after acquiring the lock in a PUT endpoint
  - do a call with postmant to that endpoint and a florence token
  - check that the lock is acquired with the florence user identity
  - do a call with postman to that endopint and a service auth token
  - check that the lock is acquired with the token owner name (as returned by zebedee)

### Who can review

anyone
